cunumeric/module.py

```diff
@@ -5577,6 +5578,519 @@ def nancumsum(
+        unary_red_code,
+        a,
+        axis=axis,
+        out=out,
+        keepdims=keepdims,
+        res_dtype=np.dtype(np.int64),
+    )
+
+
+@add_boilerplate("a")
+def nanmin(
+    a: ndarray,
+    axis: Any = None,
+    out: Union[ndarray, None] = None,
+    keepdims: bool = False,
+    initial: Optional[Union[int, float]] = None,
+    where: Any = True,
+) -> ndarray:
+    """
+    Return minimum of an array or minimum along an axis, ignoring any
+    NaNs. When all-NaN slices are encountered a RuntimeWarning is
+    raised and nan is returned for that slice except when initial
+    is set. Empty slices will raise a ValueError
+
+    Parameters
+    ----------
+    a : array_like
+        Array containing numbers whose minimum is desired. If a is not an
+        array, a conversion is attempted.
+
+    axis : {int, tuple of int, None}, optional
+        Axis or axes along which the minimum is computed. The default is to
+        compute the minimum of the flattened array.
+
+    out : ndarray, optional
+        Alternative output array in which to place the result.  Must
+        be of the same shape and buffer length as the expected output.
+        See `ufuncs-output-type` for more details.
+
+    keepdims : bool, Optional
+        If this is set to True, the axes which are reduced are left
+        in the result as dimensions with size one. With this option,
+        the result will broadcast correctly against the input array.
+
+        If the default value is passed, then `keepdims` will not be
+        passed through to the `amin` method of sub-classes of
+        `ndarray`, however any non-default value will be.  If the
+        sub-class' method does not implement `keepdims` any
+        exceptions will be raised.
+
+    initial : scalar, optional
+        The maximum value of an output element. Must be present to allow
+        computation on empty slice. See `~cunumeric.ufunc.reduce` for details.
+
+    where : array_like[bool], optional
+        Elements to compare for the minimum. See `~cunumeric.ufunc.reduce`
+        for details.
+
+    Returns
+    -------
+    nanmin : ndarray or scalar
+        Minimum of `a`. If `axis` is None, the result is a scalar value.
+        If `axis` is given, the result is an array of dimension
+        ``a.ndim - 1``.
+
+    Notes
+    -----
+    CuNumeric's implementation will not raise a Runtime Warning for
+    slices with all-NaNs
+
+    See Also
+    --------
+    numpy.nanmin, numpy.nanmax, numpy.min, numpy.max, numpy.isnan,
+    numpy.maximum
+
+    Availability
+    --------
+    Multiple GPUs, Multiple CPUs
+    """
+
+    unary_red_code = get_non_nan_unary_red_code(
+        a.dtype.kind, UnaryRedCode.NANMIN
+    )
+
+    out_array = a._perform_unary_reduction(
+        unary_red_code,
+        a,
+        axis=axis,
+        out=out,
+        keepdims=keepdims,
+        initial=initial,
+        where=where,
+    )
+
+    if a.dtype.kind == "f":
```

Comment:
can we make this recovery pass optional by putting it behind a command-line flag (say `-cunumeric:numpy-compat`)? this code uses three additional cunumeric operations that are in aggregate more expensive than the reduction itself and what they do is useless in most cases. I don't think this is the right trade-off we want to make.

Reply:
this is a slippery slope...but anyway I've introduced an environment variable for this.